src/Services/EncounterService.php

```diff
@@ -72,20 +72,38 @@ public function getUuidFields(): array
             , 'facility_location_uuid', 'billing_location_uuid'];
     }
 
+    /**
+     * Given a patient pid return the most recent patient encounter for that patient
+     * @param $pid The unique public id (pid) for the patient.
+     * @return array|null Returns the encounter if found, null otherwise
+     */
+    public function getMostRecentEncounterForPatient($pid): ?array
+    {
+        $pid = new TokenSearchField('pid', [new TokenSearchValue($pid, null)]);
+        // we discovered that most queries were ordering by encounter id which may NOT be the most recent encounter as
+        // an older historical encounter may be entered after a more recent encounter, so ordering be encounter id screws
+        // this up.
+        $result = $this->search(['pid' => $pid], true, '', ['limit' => 1, 'order' => '`date` DESC']);
```

Comment:
@bradymiller Here is the example where I use the options table.  I pass in a query limit and an order field.  I don't believe I need to escape it, but I can add that in if you'd like.

Reply:
hi @adunsulag , agree no need to escape since hard-coded.